accounts-on-demand/src/accounts_on_demand.rs

```diff
@@ -0,0 +1,228 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use dashmap::DashSet;
+use itertools::Itertools;
+use solana_client::{
+    nonblocking::rpc_client::RpcClient,
+    rpc_config::{RpcAccountInfoConfig, RpcProgramAccountsConfig},
+    rpc_filter::RpcFilterType,
+};
+use solana_lite_rpc_accounts::account_store_interface::AccountStorageInterface;
+use solana_lite_rpc_cluster_endpoints::geyser_grpc_connector::GrpcSourceConfig;
+use solana_lite_rpc_core::{
+    commitment_utils::Commitment,
+    structures::{
+        account_data::{AccountData, AccountNotificationMessage},
+        account_filter::{AccountFilter, AccountFilterType, AccountFilters},
+    },
+};
+use solana_sdk::{clock::Slot, pubkey::Pubkey};
+use tokio::sync::{broadcast::Sender, RwLock};
+
+use crate::subscription_manager::SubscriptionManger;
+
+pub struct AccountsOnDemand {
+    rpc_client: Arc<RpcClient>,
+    accounts_storage: Arc<dyn AccountStorageInterface>,
+    accounts_subscribed: Arc<DashSet<Pubkey>>,
+    program_filters: Arc<RwLock<AccountFilters>>,
+    subscription_manager: SubscriptionManger,
+}
+
+impl AccountsOnDemand {
+    pub fn new(
+        rpc_client: Arc<RpcClient>,
+        grpc_sources: Vec<GrpcSourceConfig>,
+        accounts_storage: Arc<dyn AccountStorageInterface>,
+        account_notification_sender: Sender<AccountNotificationMessage>,
+    ) -> Self {
+        Self {
+            rpc_client,
+            accounts_storage: accounts_storage.clone(),
+            accounts_subscribed: Arc::new(DashSet::new()),
+            program_filters: Arc::new(RwLock::new(vec![])),
+            subscription_manager: SubscriptionManger::new(
+                grpc_sources,
+                accounts_storage,
+                account_notification_sender,
+            ),
+        }
+    }
+
+    pub async fn reset_subscription(&self) {
```

Comment:
_reset_subscription_ is a but misleading. to me it reads like "remove all subscription" but actually what is does is _reconfigure_subscriptions_